crawl-ref/source/l-spells.cc

```diff
@@ -97,6 +98,65 @@ LUAFN(l_spells_min_range)
     PLUARET(number, spell_range(spell, 0));
 }
 
+
+/*** If this spell is aimed at (x,y), what path will it actually take?
+ * @tparam string spell name
+ * @tparam int x coordinate to aim at, in player coordinates
+ * @tparam int y coordinate to aim at, in player coordinates
+ * @treturn string|nil a list in "x1,y1 x2,y2 x3,y3..." format giving the path
+ *    the spell will take, in player coordinates
+ * @function path
+ */
+LUAFN(l_spells_path)
+{
+    spell_type spell = spell_by_name(luaL_checkstring(ls, 1), false);
+    if (!you.has_spell(spell))
+    {
+        lua_pushnil(ls);
+        return 1;
+    }
+    int power = calc_spell_power(spell, true);
+    int range = spell_range(spell, power);
+    zap_type zap = spell_to_zap(spell);
+    if( power <= 0 || range <= 0 )
+    {
+        lua_pushnil(ls);
+        return 1;
+    }
+
+    coord_def s;
+    s.x = luaL_checkint(ls, 2);
+    s.y = luaL_checkint(ls, 3);
+    coord_def aim = player2grid(s);
+
+    bolt beam;
+    beam.set_agent(&you);
+    beam.source = you.pos();
+    beam.attitude = ATT_FRIENDLY;
+    zappy(zap, power, false, beam);
+    beam.is_tracer = true;
+    beam.is_targeting = true;
+    beam.range = range;
+    beam.target = aim;
+    beam.dont_stop_player = true;
+    beam.friend_info.dont_stop = true;
+    beam.foe_info.dont_stop = true;
+    beam.ex_size = 0;
+    beam.aimed_at_spot = true;
+    beam.path_taken.clear();
+    beam.fire();
+
+    char buf[255];
+    int len = 0;
+    for (auto g : beam.path_taken) {
+        coord_def p = grid2player(g);
```

Comment:
Braces should also be on their own line.

Reply:
got it -- both style errors fixed